src/main/java/lotto/interfaces/lotto/LottoController.java

```diff
@@ -0,0 +1,69 @@
+package lotto.interfaces.lotto;
+
+import lotto.common.exception.LottoException;
+import lotto.domain.lotto.Lotto;
+import lotto.domain.lotto.LottoNumber;
+import lotto.domain.lotto.LottoService;
+import lotto.domain.lotto.WinningLotto;
+import lotto.domain.money.LottoMoney;
+import lotto.domain.money.Revenue;
+import lotto.domain.prize.LottoResult;
+import lotto.domain.ticket.LottoTicket;
+import lotto.interfaces.input.InputHandler;
+import lotto.interfaces.output.OutputHandler;
+import lotto.interfaces.prize.LottoResultResponse;
+import lotto.interfaces.ticket.LottoTicketResponse;
+
+public class LottoController {
+
+    private final LottoService lottoService;
+    private final InputHandler inputHandler;
+    private final OutputHandler outputHandler;
+
+    public LottoController(LottoService lottoService, InputHandler inputHandler, OutputHandler outputHandler) {
+        this.lottoService = lottoService;
+        this.inputHandler = inputHandler;
+        this.outputHandler = outputHandler;
+    }
+
+    public void lottoGameStart() {
+        LottoMoney lottoMoney = getLottoMoney();
+        LottoTicket lottoTicket = lottoService.generateLottoTicket(lottoMoney.getLottoCount());
+        outputHandler.printPurchasedLotto(LottoTicketResponse.from(lottoTicket));
+        WinningLotto winningLotto = getWinningLotto();
+        LottoResult lottoResult = lottoService.matchLottoTicket(lottoTicket, winningLotto);
+        outputHandler.printLottoResult(LottoResultResponse.of(lottoResult, new Revenue(lottoResult, lottoMoney)));
+    }
+
+    private LottoMoney getLottoMoney() {
+        while (true) {
+            try {
+                return inputHandler.readPurchaseAmount();
+            } catch (LottoException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
```

Comment:
while, try-catch 코드가 동일한 구조로 여러 곳에서 중복되어 쓰이고 있는 것 같아요!
저는 이번에 이 부분을 개선하기 위해 다양하게 고민해봤는데, **함수형 인터페이스**를 활용해서 코드를 간결하게 만들 수 있었습니다.
학습한 내용을 아래 블로그에 게시해두었으니 관심이 있으시다면 한 번 읽어보셔도 좋을 것 같아요! 😄 

[함수형 인터페이스와 표준 API](https://velog.io/@songsunkook/%ED%95%A8%EC%88%98%ED%98%95-%EC%9D%B8%ED%84%B0%ED%8E%98%EC%9D%B4%EC%8A%A4%EC%99%80-%ED%91%9C%EC%A4%80-API)

Reply:
맞아요! 선권님 글 읽고 많은 도움이 됐습니다. 저도 다음 미션에 적용해보도록 할게요. 좋은 글 감사합니다 👍

Reply:
```suggestion
    private LottoMoney getLottoMoney() {
        try {
            return inputHandler.readPurchaseAmount();
        } catch (LottoException e) {
            System.out.println(e.getMessage());
            return getLottoMoney();
        }
    }
```

이렇게 하면 while문을 안 써도 간결하게 작성 가능합니다!